### PR TITLE
Tests for: Header, Discussion, DiscussionList

### PR DESCRIPTION
**Pull request by diegotc86**
# Tests implemented
## Header
- Show correct title
- Header with no log out
- Header with log out
## Discussion
- Discussion component (only snapshot)
- Show correct title
- Show correct author
## DiscussionList
- Discussion List component (only snapshot)
- Render the correct amount of Discussion components

### CODE REVIEW

File: src/components/header.test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+});
+
+test("Header with no log out", () => {
+  const { asFragment, queryByRole } = render(
+    <Header title="Test-Title" hasLogOut={false} />
+  );
+  expect(asFragment()).toMatchSnapshot();
+
+  const logOut = queryByRole("button");
+  expect(logOut).toBeNull();
```
Comment (Linzeur) [inline src/components/header.test.js]: I see you have read the documentation about the different functions to validate tests. I like that. Good work Diego

File: src/components/discussion.test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  const title = getByLabelText("Discussion Title");
+  const textTitle = getNodeText(title);
+  expect(textTitle).toBe(discussion.title);
+});
+
+test("Show correct author", () => {
+  const { getByLabelText } = render(<Discussion discussion={discussion} />);
+
+  const author = getByLabelText("Discussion Author");
+  const authorText = getNodeText(author);
```
Comment (Linzeur) [inline src/components/discussion.test.js]: I get the mean of this function, could you tell me what was the reason which you had to use that?

Comment (diegotc86) [reply src/components/discussion.test.js]: Testing Library doesn't give the function to get the `innerText` of a HTML Element. But they have the `getNodeText` helper which fulfill the same objective. You can read more here:
https://testing-library.com/docs/dom-testing-library/api-helpers

File: src/components/header.test.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+});
+
+test("Header with log out", () => {
+  const { asFragment, getByRole } = render(
+    <Header title={"Test-Title"} hasLogOut={true} />
+  );
+  expect(asFragment()).toMatchSnapshot();
+
+  const logOut = getByRole("button");
```
Comment (byteconde) [inline src/components/header.test.js]: Good job Diego, thanks.

**Pull request merged**